tools/ADOLC/CMakeLists.txt

```diff
@@ -26,17 +26,29 @@ function(build_adolc EXE_NAME)
 	target_link_libraries("Tools-ADOLC-${EXE_NAME}" "${CMAKE_CURRENT_BINARY_DIR}/adol-c/adolc.lib")
 endfunction(build_adolc)
 
+set(ADOLC_EIGEN_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../ADOLC-Eigen)
+function(build_adolc_eigen EXE_NAME)
+	set(ADOLC_PREV_OUTPUT_DIRECTORY CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+	set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${ADOLC_EIGEN_OUTPUT_DIRECTORY})
+	add_executable("Tools-ADOLCEigen-${EXE_NAME}" "main.cpp" "${ADOLC_INCLUDE_DIR}/adolc/adolc.h" "${CMAKE_SOURCE_DIR}/src/cpp/shared/utils.cpp")
+	target_compile_definitions("Tools-ADOLCEigen-${EXE_NAME}" PUBLIC "DO_EIGEN" ${ARGN})
+	target_link_libraries("Tools-ADOLCEigen-${EXE_NAME}" "${CMAKE_CURRENT_BINARY_DIR}/adol-c/adolc.lib")
+	set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ADOLC_PREV_OUTPUT_DIRECTORY)
+	unset(ADOLC_PREV_OUTPUT_DIRECTORY)
+endfunction(build_adolc_eigen)
+
 build_adolc("GMM-FULL" "DO_GMM_FULL" "DO_CPP")
 build_adolc("GMM-SPLIT" "DO_GMM_SPLIT" "DO_CPP")
 
 build_adolc("BA" "DO_BA_BLOCK" "DO_CPP")
-# build_adolc("BA-Eigen" "DO_BA_BLOCK" "DO_EIGEN")
+# build_adolc_eigen("BA" "DO_BA_BLOCK")
 # TODO this doesn't work
```

Comment:
I would leave ADOLC lower in the priority list for the moment, e.g. PyTorch fixes would come before this TODO.

Reply:
My changes broke ADOLC running at all that's why I made it working